src/ORM/DataObjectSchema.php

```diff
@@ -1242,4 +1242,50 @@ protected function checkRelationClass($class, $component, $relationClass, $type)
             );
         }
     }
+
+    /**
+     * @param string|object $baseClass
+     * @param bool $includeBaseClass Whether to include fields in the base class or not
+     * @param string|array $fieldNames The field to get mappings for, for example 'HTMLText'. Can also be an array.
+     * @return array Returns an array of the fields available for the provided class and its sub-classes as follows:
+     * <code>
+     * [
+     *  'ClassName' => [
+     *      'TableName' => [
+     *          'FieldName',
+     *          'FieldName2',
+     *      ],
+     *      'TableName2' => [
+     *          'FieldName3',
+     *      ],
+     *    ],
+     * ]
+     * </code>
+     * @throws \ReflectionException
+     */
+    public static function getFieldMap($baseClass, $includeBaseClass, $fieldNames)
+    {
+        $mapping = [];
+
+        foreach (ClassInfo::subclassesFor($baseClass, $includeBaseClass) as $class) {
+            /** @var DataObject $fields */
+            $fields = $class::singleton()->getSchema()->fieldSpecs($class);
```

Comment:
```suggestion
            $fields = singleton($class)->getSchema()->fieldSpecs($class);
```

I'm not against the way this was done. `singleton` does allow you to pass the expected class as a parameter instead which provides a slightly nicer syntax.

Doing some validation on the object would probably be a good thing. As it stands, exception will be thrown if some one asks for a nonsensical class, however if we could throw our own specific exception, it would provide a better developer experience.